src/storage/src/render/sources.rs

```diff
@@ -669,13 +666,11 @@ fn upsert_commands<G: Scope>(
 ) -> Collection<G, (UpsertKey, Option<Result<Row, UpsertError>>, MzOffset), Diff> {
     let mut row_buf = Row::default();
     input.map(move |result| {
-        let order = result.position;
+        let order = result.position_for_upsert;
 
         let key = match result.key {
             Some(Ok(key)) => Ok(key),
-            None => Err(UpsertError::NullKey(UpsertNullKeyError::with_partition_id(
-                result.partition,
-            ))),
+            None => Err(UpsertError::NullKey(UpsertNullKeyError)),
```

Comment:
this is also not backward compatible, if someone has an existing source thats errored...perhaps we are okay with that

That said, it would be nice to recover the access to partition here (that would require inspecting the row and getting the correct metadata column, right?

Reply:
It is backward incompatible in that if there is an errored source and after this PR it recovers we'll end up with a negative multiplicity. This doesn't sound like an issue to me but if you disagree we can dig deeper into production sources.

IMO I don't think the partition id should return in here in any shape or form because it breaks the modularity of the upsert operator.

Reply:
@petrosagg I think its worth a comment in the release channel for the release that includes this change, but as a change seems reasonable to me